include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -917,7 +1050,7 @@ namespace VectorTools
       additional_data.mapping_update_flags = (update_values | update_JxW_values);
       MatrixFree<dim, Number> matrix_free;
       matrix_free.reinit (mapping, dof, constraints,
-                          QGauss<1>(fe_degree+1), additional_data);
+                          QGauss<1>(fe_degree+2), additional_data);
```

Comment:
i does not matter where it used in other tests examples. that's not the point. The point is that when you project from quadrature points, the 99% of usage case if `fe_degree+1`. So this case should be supported OOB. Please, do not modify `project_parallel` with `std_cxx11::function` in this PR.